### ISSUE TITLE
DNS based pod discovery going to depricated.

### ISSUE THREAD

**Issue by dmxunlimit** [wso2/kubernetes-common#22]
**Description:**

DNS based pod discovery cannot be used with the readiness probe due an limitation of the product. Also cosiderring the other limitation of the dns based pod discovery. This is going to be depricated, Hence remove the same from the configuration.
 

**Steps to reproduce:**
Configure the DNS based pod discovery and the readiness probe.
* DNS 'A' record will not be created untill the pod status is move to ready.
Hence cluster initialisation will be failed. Due to the dns lookup fail and that causes the readiness probe failure.

**Comment by ruwanta** [comment]
Yes, the way DNS base discovery was designed is no longer supported at the k8s level. This approach can not be used on newer k8s.

Also,
The API Server approach also needs to be improved, so that the some unnecessary configuration parameters can be eliminated.
e.g. Username, password should not be used at all, but need to use the API_KEY instead.
The API server URL should be able to auto discover
